pymc3/step_methods/hmc/base_hmc.py

```diff
@@ -83,14 +85,18 @@ def __init__(
 
         if vars is None:
             vars = self._model.cont_vars
-        vars = inputvars(vars)
+
+        # vars = inputvars(vars)
 
         super().__init__(vars, blocked=blocked, model=model, dtype=dtype, **theano_kwargs)
 
         self.adapt_step_size = adapt_step_size
         self.Emax = Emax
         self.iter_count = 0
-        size = self._logp_dlogp_func.size
+
+        test_point = model.test_point
+        grad_vars = [test_point[v.name] for v in typefilter(model.free_RVs, continuous_types)]
```

Comment:
maybe `continuous_vars`, to make it clear it is not the gradient of anything?

Reply:
Don't see why not.